data_validation/cli_tools.py

```diff
@@ -436,316 +394,414 @@ def _configure_validate_parser(subparsers):
-        choices=["row", "column"],
-        help="Which type of custom query (row/column)",
+
+    custom_query_subparsers = custom_query_parser.add_subparsers(
+        dest="custom_query_type"
+    )
+
+    # Add arguments for custom-query row parser
+    custom_query_row_parser = custom_query_subparsers.add_parser(
+        "row", help="Run a custom query row validation"
+    )
+    _configure_custom_query_row_parser(custom_query_row_parser)
+
+    # Add arguments for custom-query column parser
+    custom_query_column_parser = custom_query_subparsers.add_parser(
+        "column", help="Run a custom query column validation"
     )
-    custom_query_parser.add_argument(
+    _configure_custom_query_column_parser(custom_query_column_parser)
+
+
+def _configure_custom_query_row_parser(custom_query_row_parser):
+    # Group optional arguments
+    optional_arguments = custom_query_row_parser.add_argument_group(
+        "optional arguments"
+    )
+    optional_arguments.add_argument(
+        "--filters",
+        "-filters",
+        help="Filters in the format source_filter:target_filter",
+    )
+    optional_arguments.add_argument(
+        "--threshold",
+        "-th",
+        type=threshold_float,
+        help="Float max threshold for percent difference",
+    )
+    optional_arguments.add_argument(
+        "--use-random-row",
+        "-rr",
+        action="store_true",
+        help="Finds a set of random rows of the first primary key supplied.",
+    )
+    optional_arguments.add_argument(
+        "--random-row-batch-size",
+        "-rbs",
+        help="Row batch size used for random row filters (default 10,000).",
+    )
+
+    # Group required arguments
+    required_arguments = custom_query_row_parser.add_argument_group(
+        "required arguments"
+    )
+    required_arguments.add_argument(
         "--source-query-file",
         "-sqf",
         required=True,
         help="File containing the source sql query",
     )
-    custom_query_parser.add_argument(
+    required_arguments.add_argument(
         "--target-query-file",
         "-tqf",
         required=True,
         help="File containing the target sql query",
     )
-    custom_query_parser.add_argument(
+    required_arguments.add_argument(
+        "--primary-keys",
+        "-pk",
+        required=True,
+        help="Comma separated list of primary key columns 'col_a,col_b'",
+    )
+
+    # Group for mutually exclusive required arguments. Either must be supplied
+    required_mutually_exclusive = required_arguments.add_mutually_exclusive_group(
+        required=True
+    )
+    required_mutually_exclusive.add_argument(
+        "--hash",
+        "-hash",
+        help="Comma separated list of columns for hashing a concatenate 'col_a,col_b' or * for all columns",
+    )
+    required_mutually_exclusive.add_argument(
+        "--concat",
+        "-concat",
+        help="Comma separated list of columns for concat 'col_a,col_b' or * for all columns",
```

Comment:
Let's add comparison-field here (and to README doc) too..it should work in theory, just hasn't been added to the CLI

Reply:
I see there is no check present for `comp-fields` in `__main__.py` for `custom-query`, should I add there too?

Reply:
Should I add this block of [code](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/data_validation/__main__.py#L196) under `validate custom-query row`?

```
      if args.comparison_fields is not None:
          comparison_fields = cli_tools.get_arg_list(
              args.comparison_fields, default_value=[]
          )
          config_manager.append_comparison_fields(
              config_manager.build_config_comparison_fields(comparison_fields)
          )
```